src/lib/moment/diff.js

```diff
@@ -36,19 +40,45 @@ export function diff (input, units, asFloat) {
     return asFloat ? output : absFloor(output);
 }
 
+function isEndOfMonth(mom) {
+    return mom.date() === daysInMonth(mom.year(), mom.month());
+}
+
+function addWithEndOfMonth(a, b, val) {
+    if (!a.isValid()) {
+        return;
+    }
+    var dur = createDuration(val, 'months');
+    var months = absRound(dur._months);
+    var dayInMonth = a.date();
+    var newMonth = get(a, 'Month') + months;
+    var maxDays = daysInMonth(a.year(), newMonth);
+    // if End of Month, should ignore time difference
+    if (dayInMonth > maxDays || isEndOfMonth(a)) {
+        dayInMonth = maxDays;
+        var h = b._d['get' + (a._isUTC ? 'UTC' : '') + 'Hours']();
```

Comment:
Why not to store the condition result in a const and reuse it ? such as:
```suggestion
        const utcChecked = a._isUTC ? 'UTC' : '';
        const getUtcChecked = 'get' + utcChecked;
        const setUtcChecked = 'set' + utcChecked;
        const h = b._d[getUtcChecked + 'Hours']();
```
and so on ..

Reply:
done, thanks :)

Reply:
sorry forgot to resolve it :)